### PR TITLE
Logger update

### PR DESCRIPTION
**Pull request by lior-dikstein**
## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: model_compression_toolkit/qat/keras/quantizer/quantization_builder.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def get_activation_quantizer_holder(n: common.BaseNode,
     # quantization, which in this case has an empty list).
     if len(activation_quantizers) == 1:
         return KerasActivationQuantizationHolder(activation_quantizers[0])
-    Logger.error(f'KerasActivationQuantizationHolder supports a single quantizer but {len(activation_quantizers)} quantizers were found for node {n}')
+    Logger.critical(f'KerasActivationQuantizationHolder supports a single quantizer but {len(activation_quantizers)} quantizers were found for node {n}')
```
Comment (eladc-git) [inline model_compression_toolkit/qat/keras/quantizer/quantization_builder.py:52]: dot

File: model_compression_toolkit/target_platform_capabilities/target_platform/op_quantization_config.py [hunk 2]
```diff
@@ -215,24 +215,19 @@ def __init__(self,
...
-                                                        f'but found an object type: {type(cfg)}'
+            assert isinstance(cfg, OpQuantizationConfig), f'Each option must be an instance of \'OpQuantizationConfig\', but found an object of type: {type(cfg)}.'
         self.quantization_config_list = quantization_config_list
         if len(quantization_config_list) > 1:
-            assert base_config is not None, f'When quantization config options contains more than one configuration, ' \
-                                            f'a base_config must be passed for non-mixed-precision optimization process'
-            assert base_config in quantization_config_list, f"base_config must be in the given quantization config " \
-                                                            f"list of options"
+            assert base_config is not None, f'For multiple configurations, a \'base_config\' is required for non-mixed-precision optimization.'
+            assert base_config in quantization_config_list, f"\'base_config\' must be included in the quantization config options list."
```
Comment (eladc-git) [inline model_compression_toolkit/target_platform_capabilities/target_platform/op_quantization_config.py:225]: logger?

File: model_compression_toolkit/target_platform_capabilities/target_platform/target_platform_model.py [hunk 3]
```diff
@@ -192,7 +192,7 @@ def __validate_model(self):
         """
         opsets_names = [op.name for op in self.operator_set]
         if (len(set(opsets_names)) != len(opsets_names)):
-            Logger.error(f'OperatorsSet must have unique names')
+            Logger.critical(f'OperatorsSet must have unique names.')
```
Comment (eladc-git) [inline model_compression_toolkit/target_platform_capabilities/target_platform/target_platform_model.py:195]: OperatorsSets

File: model_compression_toolkit/target_platform_capabilities/target_platform/targetplatform2framework/attribute_filter.py [hunk 4]
```diff
@@ -87,7 +87,7 @@ def __or__(self, other: Any):
         """
 
         if not isinstance(other, AttributeFilter):
-            Logger.error("Not an attribute filter. Can not run an OR operation.")  # pragma: no cover
+            Logger.critical("Not an attribute filter. Cannot perform an 'OR' operation.")  # pragma: no cover
```
Comment (eladc-git) [inline model_compression_toolkit/target_platform_capabilities/target_platform/targetplatform2framework/attribute_filter.py:90]: specify 'other' in the error

File: model_compression_toolkit/target_platform_capabilities/target_platform/targetplatform2framework/attribute_filter.py [hunk 5]
```diff
@@ -101,7 +101,7 @@ def __and__(self, other: Any):
             AndAttributeFilter that filters with AND between the current AttributeFilter and the passed AttributeFilter.
         """
         if not isinstance(other, AttributeFilter):
-            Logger.error("Not an attribute filter. Can not run an AND operation.")  # pragma: no cover
+            Logger.critical("Not an attribute filter. Can not perform an 'AND' operation.")  # pragma: no cover
```
Comment (eladc-git) [inline model_compression_toolkit/target_platform_capabilities/target_platform/targetplatform2framework/attribute_filter.py:104]: specify 'other' in the error

File: model_compression_toolkit/target_platform_capabilities/target_platform/targetplatform2framework/current_tpc.py [hunk 6]
```diff
@@ -38,7 +39,7 @@ def get(self):
 
         """
         if self.tpc is None:
-            raise Exception('TargetPlatformCapabilities is not initialized.')
+            Logger.critical("'TargetPlatformCapabilities' instance is not initialized.")
```
Comment (eladc-git) [inline model_compression_toolkit/target_platform_capabilities/target_platform/targetplatform2framework/current_tpc.py:42]: I would add (TPC)

File: model_compression_toolkit/trainable_infrastructure/common/get_quantizers.py [hunk 7]
```diff
@@ -54,9 +54,9 @@ def get_trainable_quantizer_class(quant_target: QuantizationTarget,
...
 
     if len(filtered_quantizers) != 1:
-        Logger.error(f"Found {len(filtered_quantizers)} quantizer for target {quant_target.value} "  # pragma: no cover
-                     f"that matches the requested quantization method {quant_method.name} and "
-                     f"quantizer type {quantizer_id.value} but there should be exactly one."
-                     f"The possible quantizers that were found are {filtered_quantizers}.")
+        Logger.critical(f"Found {len(filtered_quantizers)} quantizers for target {quant_target.value}, "
+                        f"matching the requested quantization method {quant_method.name} and "
+                        f"quantizer type {quantizer_id.value}, but exactly one is required. "
+                        f"Identified quantizers: {filtered_quantizers}.")
```
Comment (eladc-git) [inline model_compression_toolkit/trainable_infrastructure/common/get_quantizers.py:60]: Supported instead of Identified

**Review by eladc-git — COMMENTED** [review]
Very good. I added some comments. 'dot' means missing '.' at the end of the sentence.

File: model_compression_toolkit/core/pytorch/kpi_data_facade.py [hunk 8]
```diff
@@ -92,5 +91,5 @@ def pytorch_kpi_data(in_model: Module,
     # If torch is not installed,
     # we raise an exception when trying to use this function.
     def pytorch_kpi_data(*args, **kwargs):
-        Logger.critical('Installing torch is mandatory when using pytorch_kpi_data. '
-                        'Could not find Tensorflow package.')  # pragma: no cover
+        Logger.critical("PyTorch must be installed to use pytorch_kpi_data. The 'torch' package is missing.")  # pragma: no cover
```
[the remark this answers is not in the document]
Comment (lior-dikstein) [reply model_compression_toolkit/core/pytorch/kpi_data_facade.py:92]: Fixed this throughout the code

**Review by eladc-git — APPROVED** [review]
